### PR TITLE
Add GET endpoint for RdRp geo data

### PR DESCRIPTION
**Pull request by lukepereira**
This PR is related to the feature request "RdRp Geo map - Switch from static CSV to API for Map Data": https://github.com/serratus-bio/serratus.io/issues/225.

It adds updates to an existing pending PR:  https://github.com/serratus-bio/serratus-summary-api/pull/34
    - improve consistency with codebase
    - remove redundant error checking on pagination
    - cache results
    - use `match` naming, since it's likely the geo api will support similar forms of filtering queries (see [suggestions on slack](https://hackseq-rna.slack.com/archives/C010RBKH0SG/p1675885426454489)) 

I will test this change with the front-end to make sure async batched requests aren't too slow, then decide if we need to create a new materialized view with denormalized data instead of running the current query which has two inner joins. Note, the inner joins shouldn't be too problematic if they're joining on indexed columns.

Tested locally:
- `bash run.sh` and http://127.0.0.1:5000/matches/rdrp/geo/paged?page=1&perPage=1000
- `bash test.sh` passes

### CODE REVIEW

**Comment by ababaian** [comment]
I think I noted this in the other thread somewhere; can you seperate the files for these GET endpoints to their own `geo` files? This is consistent with the rest of the project where each "module" has it's own file (nucleotide / rdrp )

`rdrp.py` --> `geo.py`
`test_rdrp.py` --> `test_geo.py`

Add approproriate imports where needed.

**Comment by lukepereira** [comment]
@ababaian Will do. Also, let me know if you have a preference for the route. I'm currently going with `/matches/rdrp/geo/paged?page=1&perPage=1000` since it seemed to make sense as a searchType in the existing `/matches/rdrp/{searchType}` code. But we could also make it's own base route, i.e. `/geo/rdrp/paged`.

**Comment by ababaian** [comment]
> @ababaian Will do. Also, let me know if you have a preference for the route. I'm currently going with `/matches/rdrp/geo/paged` since it seemed to make sense as a searchType in the existing `/matches/rdrp/{searchType}` code. But we could also make it's own base route, i.e. `/geo/rdrp/paged`.

I think `/geo/` base makes sense the most, it frames the problem from a geo starting point, which means if we add further modules like `/nucleotide/` or `/structure/` in the future, they each equivalently could use `/geo/`

**Comment by lukepereira** [comment]
FYI, I'm hoping to test out the production caching behavior before releasing the front-end to users. From what I've read and seen on localhost is that it's using a 'simple' filesystem cache with unlimited timeout but it's hard to know the behavior when running on EBS. LMK If you suspect any issues

File: model/tables/sra.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+from dataclasses import dataclass
```
[the remark this answers is not in the document]
Comment (lukepereira) [reply model/tables/sra.py:1]: The geo query doesn't actually select any of the additional fields so they are just a reference to the db model, we only select the `run`, `release_date`, `tax_id`, `scientific_name` for the geo data so there's no impact on I/O.

The additional fields could be useful for future work, but if it's just bloat and makes readability worse i will remove it

[the remark this answers is not in the document]
Comment (ababaian) [reply model/tables/sra.py:1]: ah, then I guess full compatibility will save us time in the future. I like it :+1:

**Pull request merged**